packages/teams-js/src/private/filesExtensionsFor3PStorageproviders.ts

```diff
@@ -0,0 +1,194 @@
+  }
+  export interface FileResult {
+    /**
+     * error encountered in getDragAndDropFiles API
+     */
+    error?: SdkError;
+    /**
+     * File chunk which will be assemebled and converted into a blob
+     */
+    fileChunk: FileChunk;
+    /**
+     * File index of the file for which chunk data is getting recieved
+     */
+    fileIndex: number;
+    /**
+     * File type which is getting recieved
+     */
+    fileType: string;
+    /**
+     * Will tell if this is the last file
+     */
+    isLastFile: boolean;
+    /**
+     * Will tell the name of the file
+     */
+    fileName: string;
+  }
+  export interface AttachmentListHelper {
+    fileType: string;
+    assembleAttachment: AssembleAttachment[];
+  }
+  export interface AssembleAttachment {
+    /** A number representing the sequence of the attachment in the file chunks. */
+    sequence: number;
+    /** A Blob object representing the data of the file chunks. */
+    file: Blob;
+  }
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param dragAndDropInput - Input for drag-and-drop action.
+   * @param callback - Callback function to handle the result.
+   */
+  export function getDragAndDropFilesHandler(
+    dragAndDropInput: string,
+    callback: (attachments: FilesFor3PApps[], error?: SdkError) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[getDragAndDropFiles] Callback cannot be null');
+    }
+
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+
+    if (!dragAndDropInput || dragAndDropInput === '') {
+      //condition changed
+      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      callback([], invalidInput);
+      return;
+    }
+    getFilesDragAndDropViaCallback(dragAndDropInput, callback);
+  }
+
+  function getFilesDragAndDropViaCallback(
+    dragAndDropInput: string,
+    callback: (attachments: FilesFor3PApps[], error?: SdkError) => void,
+  ): void {
+    const files: FilesFor3PApps[] = [];
+    let helper: filesExtensionsFor3PStorageproviders.AttachmentListHelper = {
+      fileType: '',
+      assembleAttachment: [],
+    };
+
+    function handleGetDragAndDropFilesCallbackRequest(
+      fileResult: filesExtensionsFor3PStorageproviders.FileResult,
+    ): void {
+      if (callback) {
+        if (fileResult && fileResult.error) {
+          callback([], fileResult.error);
+        } else {
+          if (fileResult && fileResult.fileChunk) {
+            try {
+              const assemble: filesExtensionsFor3PStorageproviders.AssembleAttachment = decodeAttachment(
+                fileResult.fileChunk,
+                fileResult.fileType,
+              );
+              helper.assembleAttachment.push(assemble);
+
+              if (fileResult.fileChunk.chunkSequence == Number.MAX_SAFE_INTEGER) {
+                const fileBlob = createFile(helper.assembleAttachment, fileResult.fileType);
+
+                if (fileResult.isLastFile) {
```

Comment:
Are we only going to check if the `chunkSequence` reaches max_int? What if it finishes before the max_int?

Reply:
From TMP code we are sending the last chunk sequence as max_int.
This will help us to identify the last chunk and sort it based on chunk sequence

Reply:
What if the file is large?